cmd/kaniko-acr/main.go

```diff
@@ -649,6 +667,25 @@ func getPublicUrl(token, registryUrl, subscriptionId string) (string, error) {
 	return "", errors.New("did not receive any registry information from /subscriptions API")
 }
 
+func setDockerAuth(username, password, registry, dockerUsername, dockerPassword, dockerRegistry string) error {
+	dockerConfig := docker.NewConfig()
+	pushToRegistryCreds := docker.RegistryCredentials{
+		Registry: registry,
+		Username: username,
+		Password: password,
+	}
+
+	pullFromRegistryCreds := docker.RegistryCredentials{
+		Registry: dockerRegistry,
+		Username: dockerUsername,
+		Password: dockerPassword,
+	}
+
+	credentials := []docker.RegistryCredentials{pushToRegistryCreds, pullFromRegistryCreds}
+	dockerConfig.CreateDockerConfig(credentials, dockerConfigPath)
```

Comment:
we should return the error from `dockerConfig.CreateDockerConfig(credentials, dockerConfigPath)`